src/app/public/modules/popover/popover.directive.ts

```diff
@@ -132,80 +133,98 @@ export class SkyPopoverDirective implements OnChanges, OnDestroy {
     }
   }
 
-  private isPopoverOpen(): boolean {
-    return (this.skyPopover && this.skyPopover.isOpen);
-  }
-
   private addEventListeners(): void {
     const element = this.elementRef.nativeElement;
 
     this.skyPopoverMessageStream
-      .takeUntil(this.idled)
+      .takeUntil(this.ngUnsubscribe)
       .subscribe(message => {
         this.handleIncomingMessages(message);
       });
 
+    Observable
+      .fromEvent(element, 'keydown')
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((event: KeyboardEvent) => {
+        if (!this.skyPopover.isActive) {
+          return;
+        }
+
+        const key = event.key.toLowerCase();
+
+        /* tslint:disable-next-line:switch-default */
+        switch (key) {
+          case 'arrowdown':
+          case 'arrowleft':
+          case 'arrowright':
+          case 'arrowup':
+          case 'down':
+          case 'left':
+          case 'right':
+          case 'up':
+            this.sendMessage(SkyPopoverMessageType.Focus);
+            event.stopPropagation();
+            event.preventDefault();
+            break;
+          case 'tab':
+            this.sendMessage(SkyPopoverMessageType.Close);
+            break;
+        }
+      });
+
     Observable
       .fromEvent(element, 'keyup')
-      .takeUntil(this.idled)
+      .takeUntil(this.ngUnsubscribe)
       .subscribe((event: KeyboardEvent) => {
         const key = event.key.toLowerCase();
-        if (key === 'escape' && this.isPopoverOpen()) {
-          event.stopPropagation();
-          event.preventDefault();
```

Comment:
If we remove `stopPropagation`, what will happen if you hit `escape` on a popover that's inside of a modal? Will it close the modal now that the event bubbles?

Reply:
Hm, you're right. I've re-added these.